tests/common/test_event.py

```diff
@@ -270,107 +298,81 @@ def test_periodic_emits_after_elapsed_delta(self, mock_event):
 
     @patch("azurelinuxagent.common.event.datetime")
     @patch('azurelinuxagent.common.event.EventLogger.add_event')
     def test_periodic_forwards_args_default_values(self, mock_event, mock_datetime):
         event.__event_logger__.reset_periodic()
         event.add_periodic(logger.EVERY_DAY, "FauxEvent", message="FauxEventMessage")
         mock_event.assert_called_once_with("FauxEvent", op=WALAEventOperation.Unknown, is_success=True, duration=0,
-                                           version=str(CURRENT_VERSION), message="FauxEventMessage", evt_type="",
-                                           is_internal=False, log_event=True)
+                                           version=str(CURRENT_VERSION), message="FauxEventMessage", log_event=True)
 
     @patch("azurelinuxagent.common.event.EventLogger.add_event")
     def test_add_event_default_variables(self, mock_add_event):
         add_event('test', message='test event')
-        mock_add_event.assert_called_once_with('test', duration=0, evt_type='', is_internal=False, is_success=True,
-                                               log_event=True, message='test event', op=WALAEventOperation.Unknown,
+        mock_add_event.assert_called_once_with('test', duration=0, is_success=True, log_event=True,
+                                               message='test event', op=WALAEventOperation.Unknown,
                                                version=str(CURRENT_VERSION))
 
+    def test_collect_events_should_delete_event_files(self):
+        add_event(name='Event1')
+        add_event(name='Event1')
+        add_event(name='Event3')
+
+        event_files = os.listdir(self.event_dir)
+        self.assertEquals(len(event_files), 3, "Did not find all the event files that were created")
+
+        event_list = event.collect_events()
+        event_files = os.listdir(self.event_dir)
+
+        self.assertEquals(len(event_list.events), 3, "Did not collect all the events that were created")
+        self.assertEquals(len(event_files), 0, "The event files were not deleted")
+
     def test_save_event(self):
         add_event('test', message='test event')
-        self.assertTrue(len(os.listdir(self.tmp_dir)) == 1)
+        self.assertTrue(len(os.listdir(self.event_dir)) == 1)
 
         # checking the extension of the file created.
-        for filename in os.listdir(self.tmp_dir):
-            self.assertEqual(".tld", filename[-4:])
-
-    def test_save_event_message_with_non_ascii_characters(self):
-        test_data_dir = os.path.join(data_dir, "events", "collect_and_send_extension_stdout_stderror")
-        msg = ""
-
-        with open(os.path.join(test_data_dir, "dummy_stdout_with_non_ascii_characters"), mode="r+b") as stdout:
-            with open(os.path.join(test_data_dir, "dummy_stderr_with_non_ascii_characters"), mode="r+b") as stderr:
-                msg = read_output(stdout, stderr)
```

Comment:
This test was passing for the wrong reason. It beats me why we were going thru read_output to prepare the test case, but by doing so an error was happening during the call to read_output and the string in the test event was actually the error message from that exception.